### PR TITLE
Check asynchDNS

### LINKED ISSUE #105: Trigger warning when cURL AsynchDNS is not enabled

### PR DESCRIPTION
**Pull request by JanPetr**
New PR from #103, because of new target branch.

Fixes #105

### CODE REVIEW

**Comment by JanPetr** [comment]
Method `isAsynchDNSAvailable` tested on all PHP >= 5.4. Both CLI and HTML output. Works on all the versions 💯

**Comment by rayrutjes** [comment]
The work done here should be bundled in a small script that allows to check the requirements.
Some kind of "benchmark" script would be great.

I'm still not convinced of the reliability of this, and we can not rely on a function that is blocked for security reasons in some environments.

File: src/AlgoliaSearch/Client.php [hunk 1]
```diff
@@ -1032,14 +1038,39 @@ private function getPlacesIndex()
...
+    /**
+     * Checks if the server has enabled AsynchDNS feature
+     * The feature is required for correct DNS fallbacks
+     *
+     * @return bool
+     */
+    private function isAsynchDNSAvailable()
+    {
+        ob_start();
+        phpinfo();
```
Comment (lt) [inline src/AlgoliaSearch/Client.php]: `curl.h` contains
```c
#define CURL_VERSION_ASYNCHDNS    (1<<7)  /* Asynchronous DNS resolves */
```

In theory something like this should work, but I haven't tested.

```php
$curlVersion = curl_version();
$hasAsynchDNS = $curlVersion['features'] & (1 << 7);
```

Comment (rayrutjes) [inline src/AlgoliaSearch/Client.php]: Thanks for that @lt. 
We will test this the moment we decide to move on with this detection.

[the remark this answers is not in the document]
Comment (julienbourdeau) [reply src/AlgoliaSearch/Client.php]: Thanks @lt for your inputs 🙏 very clever! 
I opened https://github.com/algolia/algoliasearch-client-php/pull/464 to port this to the client v2.

**Pull request closed without merging**